packages/react-components/theme-designer/src/ThemeDesigner.state.ts

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import {
+  createDarkTheme,
+  createLightTheme,
+  teamsDarkTheme,
+  teamsLightTheme,
+  webDarkTheme,
+  webLightTheme,
+} from '@fluentui/react-components';
+import { getBrandTokensFromPalette } from './utils/getBrandTokensFromPalette';
+import { brandTeams, brandWeb } from './utils/brandColors';
+
+import type { BrandVariants, Theme } from '@fluentui/react-components';
+
+export type CustomAttributes = {
+  keyColor: string;
+  hueTorsion: number;
+  darkCp: number;
+  lightCp: number;
+  isDark: boolean;
+};
+
+export type DispatchTheme = React.Dispatch<{
+  type: string;
+  customAttributes: CustomAttributes;
+}>;
+
+export const useThemeDesignerState = () => {
```

Comment:
lets change this to useThemeDesignerReducer since we are returning the result of a reducer